packages/devtools_app/lib/src/preferences.dart

```diff
@@ -60,4 +68,34 @@ class PreferencesController {
   void toggleDenseMode(bool enableDenseMode) {
     _denseMode.value = enableDenseMode;
   }
+
+  /// Get the value for the split fractions.
+  ValueNotifier<List<double>> lookupSplitFractions(String key) {
+    return splitFractions.putIfAbsent(
+        key, () => ValueNotifier<List<double>>(_initialFractions[key] ?? []));
+  }
+
+  /// Update the value for the split fractions.
+  void updateSplitFractions(String key, List<double> value) {
+    if ((key?.isEmpty ?? true) ||
+        (value?.length ?? 0) < 2 ||
+        isListEqual(lookupSplitFractions(key).value, value)) return;
+    lookupSplitFractions(key).value = value;
+    storage?.setValue(
+        'ui.splitFractions', jsonEncode(fractionsToJson(splitFractions)));
+  }
+
+  Map<String, List<double>> fractionsFromJson(Map<String, dynamic> json) {
+    return json
+        .map((key, fractions) => MapEntry(key, fractions?.cast<double>()));
```

Comment:
`fractions?.cast<double>()` will yield null when `fractions` is null. Instead of null, should we be using an empty list?